tests/scripts/check-generated-files.sh

```diff
@@ -11,14 +17,52 @@ fi
 
 check()
 {
-    FILE=$1
-    SCRIPT=$2
+    SCRIPT=$1
+    TO_CHECK=$2
+    PATTERN=""
+
+    if [ -d $TO_CHECK ]; then
+        for FILE in $TO_CHECK/*; do
+            FILES="$FILE $FILES"
+        done
+    else
+        FILES=$TO_CHECK
+    fi
+
+    for FILE in $FILES; do
+        cp $FILE $FILE.bak
+    done
 
-    cp $FILE $FILE.bak
     $SCRIPT
-    diff $FILE $FILE.bak
-    mv $FILE.bak $FILE
+
+    # Compare the script output to the old files and remove backups
+    for FILE in $FILES; do
+        if ! diff $FILE $FILE.bak >/dev/null 2>&1; then
+            echo "'$FILE' was either modified or deleted by '$SCRIPT'"
+            exit 1
```

Comment:
Exiting here, leaves all the intermediate `*.bak` files behind. Do we really want to do that?

Reply:
I personally do not mind either way. I left it like this because the behaviour of check-generated-files.sh was already like this anyways. Would you like me to implement this change?

Reply:
I think it would be better to clear up the files, but if this is what the other scripts do, we've already set a low bar and this is just a test script, mainly for use on the CI which cleans up after itself, so I think your time is better spent elsewhere. Forget it.